src/components/HomePage/index.js

```diff
@@ -14,12 +14,27 @@ import useStyles from "./styles";
 
 function HomePage({ background = "white", textColor = "black" }) {
   const classes = useStyles();
+  const trendingTags = ["javascript", "react", "html", "css"];
   const [value, setValue] = useState(2);
+  const [posts, setPosts] = useState(userList);
   const handleChange = (event, newValue) => {
     setValue(newValue);
   };
+
+  const handleClick = (e) => {
+    let updatedPosts = userList.persons;
+    updatedPosts = updatedPosts.filter((post) =>
+      post.tags.includes(e.target.innerHTML)
```

Comment:
avoid using `innerHTML`. Instead, add pass the item to handle click and then do the filtering